src/LocalInstaller.ts

```diff
@@ -24,14 +25,16 @@ export interface ListByPackage {
 }
 
 export class LocalInstaller extends EventEmitter {
-
     private sourcesByTarget: ListByPackage;
     private options: Options;
+    private uniqueDir: string;
 
-    constructor(sourcesByTarget: ListByPackage, options?: Options) {
+    constructor(sourcesByTarget: ListByPackage, options?: Options, uniqueDir?: string) {
```

Comment:
Could we instead make a private constructor argument here: `private uniqueDir = uniqueid('node-local-install-')`?

Reply:
I don't really see the point in clogging up the constructor like this while the current approach works just fine. Especially because the other private variables are defined the same way.

Reply:
The reason that the other attributes cannot be set like that is that they are not provided to the constructor, but rather calculated in the constructor. For optional variables, I much prefer the optional default parameter syntax.